File: 1726-tuple-with-same-product/1726-tuple-with-same-product.cpp
class Solution {
public:
  int tupleSameProduct(vector<int> &nums) {
    int n = (int)nums.size();

    map<int, int> mp;
    for (int i = 0; i < n; i++) {
      for (int j = i + 1; j < n; j++)
        mp[nums[i] * nums[j]]++;
    }

    int ans = 0;
    for (auto [val, fr] : mp)
      ans += (((fr * (fr - 1)) / 2) * 8);

    return ans;
  }
};

File: 1726-tuple-with-same-product/1726-tuple-with-same-product.py
class Solution:
    def tupleSameProduct(self, nums: List[int]) -> int:
        freq = {}
        sz = len(nums)

        for i in range(0,sz):
            for j in range(i+1,sz):
                x = nums[i]
                y = nums[j]
                if x != y:
                    prod = x * y
                    if prod in freq:
                        freq[prod] += 1
                    else:
                        freq[prod] = 1

        ans = int(0)
        for key, val in freq.items():
            x = int((val * (val - 1)) / 2)
            ans += x * 8
        
        return ans
